nixos/modules/services/networking/yggdrasil.nix

```diff
@@ -0,0 +1,119 @@
+          started, resulting in a random IPv6 address on the yggdrasil
+          network each time.
+
+          If this option is set, <option>config</option> will be ignored.
+        '';
+      };
+
+      config = mkOption {
+        type = str;
+        default = "";
+        description = ''
+          JSON or HJSON configuration for yggdrasil.
+
+          Warning: this is stored in the WORLD-READABLE Nix store!
+          Therefore, it is not appropriate for private keys.  If you
+          provide a partial configuration and do not specify the keys,
+          yggdrasil will generate a new set each time the service is
+          started, creating a random IPv6 address on the yggdrasil
+          network each time.
+
+          If you wish to specify the keys, use
+          <option>configFile</option>, which takes precedence over
+          <option>config</option>.
+        '';
+      };
+
+      port = mkOption {
+        type = nullOr int;
+        default = null;
+        example = 54321;
+           description = ''
+          Firewall port to open in order to permit incoming peerings.
+          This should match the port specified in <code>Listen</code>
+          in the yggdrasil configuration.  It is not necessary to open
+          a port in order to make outgoing peerings.
+        '';
+      };
+
+      openMulticastPort = mkOption {
+        type = bool;
+        default = true;
+        description = ''
+          Whether to open the UDP port used for multicast peer discovery
+          on local networks.
+        '';
+      };
+
+      package = mkOption {
+        type = package;
+        default = pkgs.yggdrasil;
+        defaultText = "pkgs.yggdrasil";
+        description = "Yggdrasil package to use.";
+      };
+
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.yggdrasil.configFile = mkDefault (
+      if cfg.config != "" then
+        toString(pkgs.writeTextFile {
+          name = "yggdrasil-conf";
+          text = cfg.config;
+        })
+      else null);
+
+    systemd.services.yggdrasil = {
+      description = "Yggdrasil Network Service";
+      path = [ cfg.package ];
+      bindsTo = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      preStart = ''
+        mkdir -p /var/lib/yggdrasil/
+        chmod 700 /var/lib/yggdrasil
+        chown -R root:root /var/lib/yggdrasil
+        '' + (if (cfg.configFile == null) then ''
+           ${cfg.package}/bin/yggdrasil -genconf > /var/lib/yggdrasil/yggdrasil.conf
+        '' else ''
+          ${cfg.package}/bin/yggdrasil -normaliseconf -useconffile \
+            ${cfg.configFile}  > /var/lib/yggdrasil/yggdrasil.conf
+        '');
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/yggdrasil -useconffile /var/lib/yggdrasil/yggdrasil.conf";
+        ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
```

Comment:
In packages pkgs. is normally not referenced via pkgs. instead we import the needed dependency directly.
So in this case, we would remove the pkgs from the arguments in line 1 and change it to coretutils

Reply:
ah nvm, my search confused me and I ended up here :D